core/src/main/java/io/grpc/transport/ServerTransportListener.java

```diff
@@ -48,4 +49,9 @@
    */
   ServerStreamListener streamCreated(ServerStream stream, String method,
       Metadata.Headers headers);
+
+  /**
+   * The transport completed shutting down. All resources have been released.
+   */
+  void transportTerminated();
```

Comment:
Ok thanks for the clarification ... that's what I had originally suspected but was derailed somewhere though the discussion :)

I guess I'm not sure what the race condition is that you're concerned about by using a single state. Do you have an example in mind?